example-library/src/main/java/uk/gov/gchq/palisade/example/library/config/ExampleUserConfiguration.java

```diff
@@ -50,19 +50,19 @@ public ExampleUserConfiguration() {
      * @param users a list of objects implementing the {@link uk.gov.gchq.palisade.service.user.config.UserPrepopulationFactory} interface
      */
     public ExampleUserConfiguration(final List<ExampleUserPrepopulationFactory> users) {
-        this.users = users;
+        this.users = List.copyOf(users);
```

Comment:
So we want the instance parameter (via the setter and constructor) to be an immutable set and the getters to provide an unmodifiable set.    What is the advantage of this?

Reply:
The goal is to prevent unexpected changes if you did:
```java
a = List.of(1,2)
b.setA(a)
a.append(3)
print(b.getA()) // [1, 2], not [1,2,3]
```
or
```java
a = List.of(1,2)
b.setA(a)
a2 = b.getA()
a2.append(3)
print(b.getA()) // [1,2] not [1,2,3]
```

Reply:
We want to ensure:
* any input list is a copy
* any output list is a copy, possibly immutable

Reply:
Getters have been made to use the immutable return i.e. `List.copyOf()`
Setters have been made to use the unmodifiable value i.e. `Collections.unmodifiableList()`